rllib/utils/checkpoints.py

```diff
@@ -208,7 +208,11 @@ def convert_to_msgpack_checkpoint(
     # Convert all code in state into serializable data.
```

Comment:
Could we split this function into two? or be able to provide an Algorithm instance, via a parameter? Sometimes it is not possible to use Algorithm.from_checkpoint() if we have some special config. So we need to create it manually but still would like to use this function from an existing algorithm instance.

Reply:
I'm not sure I understand. This function should only be used once on n existing (pickle-based) checkpoint directories that you pass in as args. Then the function spits out n new msgpack-based checkpoints. You can then use these new msgpack checkpoints just like any other checkpoint (pickle-based) for restoring an Algorithm in different ways:
* Via `tune.Tuner.restore()`
* Via the classmethod: `new_algo = Algorithm.from_checkpoint()`.
* Via the "old" two-step approach: `algo = MyAlgoClass(config, ...); algo.restore([msgpack checkpoint directory])`

I hope this makes sense. If not, please let me know, what's unclear, so I can try to explain further.

Reply:
Yes, I understand the usecase where it should be used. But in my case Algorithm.from_checkpoint does not work. It crashes internally because the custom env is not setup. So basically I have to create the Algorithm manually and restore the checkpoint.